### DISCUSSION TITLE
Smartbanner for Huawei without Play Store

### DISCUSSION THREAD

**Discussion by Funky89** [ain/smartbanner.js#497] [Ideas]
Hi there,
Have you implemented or in the future the banner for Huawei phones that do not have the Play Store?
Unless you have an idea how to work around the problem ?

Best regards

Funky

**Comment by ain** [comment]
Hi @Funky89! Good question. So far you're the first one to raise the question.

I could imagine that Huawei devices would in that case be handled differently as a platform, next to iOS and Android. The question is if such a generalisation is safe and if it is always the case that all Huawei devices are bound to their dedicated, separate Huawei app store?

Additionally, how prominent is such a platform on the market? Does this feature make sense?

**Comment by Funky89** [comment]
Hello @ain , 
Currently since May 2019 all new Huawei phones no longer have Google services. 
Maybe in future years it will come back.
Right now all new Huawei phones have the AppGallery

From these data from the [site](https://gs.statcounter.com/vendor-market-share/mobile) , we can see since last year that Huawei phones are still present but with a sharp drop and [link](https://omdia.tech.informa.com/-/media/tech/omdia/marketing/pr/2022-jan/annual-result-2021-smartphone-shipments.jpg?rev=a730b122197a4d869c1ed120a13b45b2&hash=B932C5CAD1A9B086295D8C97CD238846)

> Does this feature make sense?

If Huawei sales are down, I don't think so

However, I can go with your idea and go with the verification of the HMSCore user agent and then modify all the options by adding huawei (icon, url button)

EDIT : I just tested where you return ios or android by adding a new condition to test the Huawei user agent.

It works with Huawei's default browser which does return HMSCore. On the other hand, if you want to use google Chrome, you must rely on device ids.

On Firefox impossible to detect HMSCore or the device ids

Best regards

Funky

**Comment by j-ixcayau** [comment]
Is there any update of this topic?

**Reply by ain** [reply]
Unfortunately not. We have issues like #687 that are the priority, but if someone is willing to prepare a well-structured and tested feature contribution, this could be merged.
